core/src/main/java/com/es/core/model/phone/setextractors/PhoneListResultSetExtractor.java

```diff
@@ -0,0 +1,79 @@
+package com.es.core.model.phone.setextractors;
+
+import com.es.core.model.phone.Color;
+import com.es.core.model.phone.Phone;
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.ResultSetExtractor;
+import org.springframework.stereotype.Component;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class PhoneListResultSetExtractor implements ResultSetExtractor<List<Phone>> {
+    @Override
+    public List<Phone> extractData(ResultSet resultSet) throws SQLException, DataAccessException {
+        Map<Long, Phone> phones = new HashMap<>();
+
+        while (resultSet.next()) {
+            Long id = resultSet.getLong("id");
+            Phone phone = phones.get(id);
+
+            if (phone == null) {
+                phone = getPhone(id, resultSet);
+                phones.put(id, phone);
+            }
+
+            addColorToPhone(phone, resultSet);
+        }
+
+        return new ArrayList<>(phones.values());
```

Comment:
Not sure if the order will be maintained with that approach, but that's ok for now